examples/dotnet/SeleniumDocs/Bidirectional/BidiApiTest.cs

```diff
@@ -8,32 +11,50 @@ namespace SeleniumDocs.Bidirectional
     public class BidiApiTest : BaseChromeTest
     {
         [TestMethod]
-        public async Task InterceptNetworkForAuthentication()
+        public async Task BasicAuthentication()
         {
             var handler = new NetworkAuthenticationHandler()
             {
-                UriMatcher = _ => true,
+                UriMatcher = uri => uri.AbsoluteUri.Contains("herokuapp"),
                 Credentials = new PasswordCredentials("admin", "admin")
             };
 
             INetwork networkInterceptor = driver.Manage().Network;
             networkInterceptor.AddAuthenticationHandler(handler);
-
             await networkInterceptor.StartMonitoring();
+
             driver.Navigate().GoToUrl("https://the-internet.herokuapp.com/basic_auth");
             await networkInterceptor.StopMonitoring();
 
-            Assert.AreEqual("Congratulations! You must have the proper credentials.", driver.FindElement(By.TagName("p")).Text);
+            Assert.AreEqual("Congratulations! You must have the proper credentials.",
+                driver.FindElement(By.TagName("p")).Text);
+        }
+
+        [TestMethod]
+        public async Task PinScript()
+        {
+            driver.Url = "https://www.selenium.dev/selenium/web/javascriptPage.html";
+
+            JavaScriptEngine engine = new JavaScriptEngine(driver);
+            PinnedScript isDisplayed = await engine.PinScript(IsDisplayedScript());
+
+            IWebElement hidden = driver.FindElement(By.Id("hiddenlink"));
+            IWebElement visible = driver.FindElement(By.Id("visibleSubElement"));
 
+            var isVisibleDisplayed = ((WebDriver)driver).ExecuteScript(isDisplayed, visible);
+            var isHiddenDisplayed = ((WebDriver)driver).ExecuteScript(isDisplayed, hidden);
+
+            Assert.IsTrue((bool)isVisibleDisplayed);
+            Assert.IsFalse((bool)isHiddenDisplayed);
         }
 
         [TestMethod]
         public async Task InterceptNetworkResponse()
```

Comment:
![image](https://github.com/SeleniumHQ/seleniumhq.github.io/assets/22616990/7823b2af-ee3a-4d51-8da5-10c5e460abf5)

this is how it looks for end user...

Reply:
Thanks, I'll fix